src/crimson/os/seastore/seastore.cc

```diff
@@ -205,11 +206,14 @@ seastar::future<> SeaStore::write_fsid(uuid_d new_osd_fsid)
 
 SeaStore::mkfs_ertr::future<> SeaStore::mkfs(uuid_d new_osd_fsid)
 {
-  return read_meta("mkfs_done").then([this, new_osd_fsid] (auto tuple) {
+  return read_meta("mkfs_done").then([this, new_osd_fsid] (auto tuple) mutable {
```

Comment:
i'd suggest fix the caller instead. in the case of teuthology test, i assume that the only possible caller is `main()` in `crimson/osd/main.cc`. so `osd_uuid` should have been specified at command line, right?

could you paste the pulpito link or point me to the test suite ? so i can take a look?

Reply:
I think the solution is really just to randomize a 0 fsid in SeaStore::mkfs.

Reply:
> i'd suggest fix the caller instead. in the case of teuthology test, i assume that the only possible caller is `main()` in `crimson/osd/main.cc`. so `osd_uuid` should have been specified at command line, right?
> 
> could you paste the pulpito link or point me to the test suite ? so i can take a look?

it is intel inner teuthology test environment. 
uuid is not set in command line when doing mkfs.   
 sudo MALLOC_CHECK_=3 adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage ceph-osd --no-mon-config --cluster ceph --mkfs --mkkey -i 0 --monmap /home/ubuntu/cephtest/ceph.monmap

Reply:
i see. can we do this in `main.cc` instead? IMHO, SeaStore is supposed to do whatever it is told to do. changing the input parameter, or the captured parameter in this case, could be confusing. like
```c++
if (config.count("mkfs")) {
  auto osd_uuid = local_conf().get_val<uuid_d>("osd_uuid");
  if (osd_uuid.is_zero()) {
    // use a random osd uuid if not specified
    osd_uuid.generate_random();
  }
  osd.invoke_on(...);
}
```

Reply:
Yeah, I agree with this change.  There's no reason for this to be store specific -- it's really a property of the behavior of the osd_uuid config.  Let's change this in main.cc as suggested here.